merge collab-nrfx-4.0 to main branch

jaz1-nordic opened the pull request:
manifest-pr-skip

masz-nordic committed:
[nrf noup] modules: hal_nordic: cleanup nrfx_config

nrf-squash! [nrf noup] modules: hal_nordic: adjust nrfx reservations to NCS

- Remove temporary workaround for SDC in nrfx's reserved resources.
- Modify NCS reserved resources to use upstream version with additions
  instead of creating new file
- Align filename

Signed-off-by: Marcin Szymczyk <marcin.szymczyk@nordicsemi.no>

nandojve committed:
[nrf fromtree] west.yml: Bump TF-M to fix FOTA upgrade regression

Update TF-M to include fixes restoring FOTA upgrade compatibility
between Zephyr v3.7.1 and v4.3.0. This fix allow both forward and
rollback upgrades.

Signed-off-by: BUDKE Gerson Fernando <gerson.budke@leica-geosystems.com>
(cherry picked from commit 9a455998a4ffae941f1023e32e1f29f7cb43449e)

standalone-tech committed:
[nrf fromtree] drivers: i2s: nrfx: Allow MCK bypass for proper LRCK and MCK Ratios

nRF53 series SoCs have a dedicated configurable audio PLL and the ability
to enable MCK bypass via a register value CONFIG.CLKCONFIG. This can
enable higher MCK/LRCK ratios that some I2S peripherals require the host
to generate. Allow an application developer to choose if they want to
initially look for a bypass ratio and, if found, enable bypass in the
NRFX driver. If not, the standard MCK calculation is conducted as normal.

Signed-off-by: Sean O'Connor <sean@standalone.tech>
(cherry picked from commit 8115d9b453ecf694e42b285821cb9ba85409a4d6)

nordicjm committed:
[nrf fromtree] drivers: Fix some Kconfig bleeds

Fixes instances of Kconfig options appearing for completely
irrelevant builds

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>
(cherry picked from commit 8498c39e13845076668aaf42513c1904a969fbb6)

mstasiaknordic committed:
[nrf fromtree] samples: nordic: nrfx_prs: remove deprecated UARTE API

nrfx_uarte_rx is deprecated and will be soon removed.
Replaced by new API.

Signed-off-by: Michał Stasiak <michal.stasiak@nordicsemi.no>
(cherry picked from commit 40f07a0532ed1ddabf33180eb2be0618bc19e832)

standalone-tech committed:
[nrf fromtree] samples: i2s: echo: Enable nRF53 MCK output in I2S sample

Enable the MCK output to a pin to allow for DAC and ADC clocking.
Provides a MCK source to peripherals that can utilize an external
clocking source for internal DSP processes, especially at higher
MCK/LRCK ratios.

Signed-off-by: Sean O'Connor <sean@standalone.tech>
(cherry picked from commit d1a7d1dde15820aca6a5222853d33c45427073ae)

nordicjm committed:
[nrf fromtree] west.yml: MCUboot synchronization from upstream

Update Zephyr fork of MCUboot to revision:
  d5b0dcb9aaee397fc105ae2228e8030038c3d871

Brings following Zephyr relevant fixes:

  - d5b0dcb9 hooks: Use dedicated type for slot numbers
  - cf138a0f boot: zephyr: config: Introduce MCUBOOT_BOOT_MAX_ALIGN
    Kconfig symbol
  - 677f1602 boot: zephyr: boards: delete frdm_mcxa156 configuration
  - 3c8e5717 zephyr: nRF54L05 and L10 configuration with LTO enabled
  - 4ea993ce zephyr: Remove duplicate zephyr module line
  - 8d14eebf boot: bootutil: Fix upgrade only serial recovery slot
    info
  - 78f87cf3 boot: zephyr: boards config of the variant stm32 disco
    kit
  - cbd3310e boot: zephyr: get base addres from dts for flexspi
    devices
  - 425e22be boot/zephyr: fix compilation errors in logging.
  - ec26273f bootutil: fix null pointer dereference in logging.
  - 3eb23a5d zephyr: Kconfig to control MCUBOOT_USE_TLV_ALLOW_LIST
  - a13624f0 bootutil: Add MCUBOOT_USE_TLV_ALLOW_LIST
  - e375252c boot: Enable Encryption with PSA + ECDSA
  - 21ed226f boot: bootutil: Fix bootutil_find_key definition
  - 763be6a5 imgtool: Add sha512 to allowed configurations for
    ECDSA256P1
  - 59d2f7a5 imgtool: Add support for VID and CID
  - 94ad4d44 boot: Add VID and CID checks

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>
(cherry picked from commit 3cd292af8c37e42634e73e2dcae04a9aa2e2152e)

wearyzen committed:
[nrf fromtree] manifest: update TF-M to fix MPS4 warnings

Use TF-M that has cherry-picked commits from upstream to fix compiler
warnings for MPS4.

Signed-off-by: Sudan Landge <sudan.landge@arm.com>
(cherry picked from commit ddb0da152179d75dddddd3eded83850a171d2550)

degjorva committed:
[nrf fromtree] manifest: Update commit id for trusted-firmware-m

Update commit id for trusted-firmware-m
to bring in support for nRF54LM20A/ns.

Signed-off-by: Dag Erik Gjørvad <dag.erik.gjorvad@nordicsemi.no>
(cherry picked from commit e151273543997b1d390ee8e660422c1e92bdf65d)

nordicjm committed:
[nrf fromtree] west.yml: MCUboot synchronization from upstream

Update Zephyr fork of MCUboot to revision:
  48b0f6da9af8d009eb8eafba023998a7d85320a1

Brings following Zephyr relevant fixes:

  - 48b0f6da imgtool: support producing images in test mode
  - 2b9f2a2f boot: zephyr: fix getting base address for certain
    boards
  - 028d80c3 bootutil: Move boot_enc_init in boot_swap_image
  - 15d43f20 zephyr: Deprecate MCUBOOT_VERIFY_IMG_ADDRESS
  - fd2c1428 zephyr: Added MCUBOOT_CHECK_HEADER_LOAD_ADDRESS
    Kconfig
  - 1035cc01 bootutil: Add MCUBOOT_CHECK_HEADER_LOAD_ADDRESS
  - 17a6ecd2 boot: bootutil: fix image_index definition
  - cc985298 boot: zephyr: Fix IO-based entrance method
  - 9ca08817 bootutil: Fix minor issues
  - 8a07053d bootutil: Fix log module registration
  - 5e1be19f boot: bootutil: write_sz fix
  - aed3fb95 docs: Add release note entry for
    MCUBOOT_BOOT_MAX_ALIGN Zephyr change

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>
(cherry picked from commit 6f41b3e2a2bd654989e98e7c0842c063f27c6da7)

nordic-krch committed:
[nrf fromtree] drivers: timer: nrf_rtc: Kconfig clean up

Remove redundant enabling of NRF_RTC_TIMER in SoC specific files
and replace it with default y in the NRF_RTC_TIMER definition.

Signed-off-by: Krzysztof Chruściński <krzysztof.chruscinski@nordicsemi.no>
(cherry picked from commit 182a6c62b140873c4ff284d13d6837fcf9029036)

aescolar committed:
[nrf fromtree] manifest: Update nRF hw models to latest

Update the HW models module to:
26ed181181eed53e400db8f63fa83c566a05d971

Including the following:
26ed181 54L CLOCK: Fix: Handle clock being stopped before it is fully
	started
b643b28 54L CLOCK: Account for relationship between PLLs and XO

Signed-off-by: Alberto Escolar Piedras <alberto.escolar.piedras@nordicsemi.no>
(cherry picked from commit 4ef316c72dbf9d228631f784c163c17260d6901f)

nordicjm committed:
[nrf fromtree] west.yml: MCUboot synchronization from upstream

Update Zephyr fork of MCUboot to revision:
  b192716c969ad358bb3a1db60c898212f3275c55

Brings following Zephyr relevant fixes:

  - b192716c Revert "loader: Allow to specify slot number in
    version"
  - 5b7bccf9 boot: bootutil: swap_offset: Fix not including
    unprotected TLVs
  - ee9efe01 bootutil: Replace boot_write_enc_key with
    boot_write_enc_keys
  - 80a39c64 bootutil: Replace literal slot indexes with
    identifiers

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>
(cherry picked from commit fbdc8b2d7bbf257804f22bd5b00673ab39e31065)

aescolar committed:
[nrf fromtree] drivers timer nrf_rtc: Fix dependency

8498c39e13845076668aaf42513c1904a969fbb6 seems to have introduce
an incorrect dependency which prevents the RTC timer from
been built if any of the RTC's are enabled.
It should only depend on the rtc1.

Signed-off-by: Alberto Escolar Piedras <alberto.escolar.piedras@nordicsemi.no>
(cherry picked from commit dd06e7ec72c0723ad9d78d9b11ee3a1702ef7b37)

tpambor committed:
[nrf fromtree] west.yml: Bump TF-M to include fixes for stm32h5

Bump TF-M to fix a linker error and address several warnings
when compiling for STM32H5.

Signed-off-by: Tim Pambor <tim.pambor@codewrights.de>
(cherry picked from commit 473d90ba6294aa7bda0a620f9327b768a88d29ab)

nordicjm committed:
[nrf fromtree] west.yml: MCUboot synchronization from upstream

Update Zephyr fork of MCUboot to revision:
  96576b341ee19f1c3af6622256b0d4f3d408e1e3

Brings following Zephyr relevant fixes:

  - 96576b34 boot: zephyr: socs: stm32h7s7xx: Add support for
    ext_flash_app variant
  - 56538cf6 bootutil: Move update-independent code
  - fccd8905 loader: Move boot_get_max_app_size(..) API
  - 0248a023 bootutil: Move state-independent area APIs
  - f2d3f00a loader: Unify image check API.
  - dcc66e51 loader: Unify header_valid(..) API
  - 257265c8 loader: Optimize boot_check_header_erased(..)
  - 76e56e4f loader: Rename boot_version_cmp
  - 5311589b loader: Fix compile-time issues in loader.c
  - 606a1934 boot: zephyr: socs: add overlay & conf for
    stm32h573xx_ext_flash_app
  - 521fc0bf bootutil: Conditionally include mbedtls/ oid.h,
    asn1.h
  - 71b41e38 boot: zephyr: boards: remove nrf54h20dk overlay
  - 2fc1bd84 bootutil: Drop slot number and boot_state from most
    boot_enc functions
  - 1dd8ae60 boot: zephyr: rework stm32h750b_dk board overlay &
    conf

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>
(cherry picked from commit cd2079f23c191558f472bc00dfd294da67e14eed)

nika-nordic committed:
[nrf fromtree] nordic: update and align to nrfx 4.0.1

New nrfx release contains major rework of nrfx drivers
instantiation making it easier to integrate with dts nodes.
Now, nrfx driver instances can no longer be `const`
because they contain driver runtime state.
Additionally, all nrfx drivers return `errno` error codes
instead of deprecated `nrfx_err_t`.

Signed-off-by: Nikodem Kastelik <nikodem.kastelik@nordicsemi.no>
(cherry picked from commit ad1e5ac2538c13f855c0591d164b9b434b2fd129)

nordicjm committed:
[nrf fromtree] west.yml: MCUboot synchronization from upstream

Update Zephyr fork of MCUboot to revision:
  f3cc9476e233364031e9ab842290392f260fba82

Brings following Zephyr relevant fixes:

  - f3cc9476 updates for 2.3.0-rc1 release
  - 496f74f3 zephyr: Fix BOOT_DOWNGRADE_PREVENTION_CHOICE symbol
  - 6ab9afc6 boot_serial: Use boot_state_init where intent to
    initialize
  - e6fefac2 bootutil: Temporarly drop mem cleanup from
    boot_state_clear
  - 457be0cf bootutil: Fix some debug log format specifiers
  - f9ad3ee2 bootutil: Add API to lock HW counter
  - 6e602004 boot: bootutil: bootutil_area: Add debug for adjusted
    offset
  - 9d334f9b boot: bootutil: bootutil_area: Fix swap using offset
    min scramble
  - 76036133 boot: zephyr: Remove enable from beginning of prompts
  - d14ba22d bootutil: Fix encryption context de initialization in
    boot_state_clear
  - d8c4cc69 bootutil: Remove NULL state logic from boot_state_clear
  - 8ff6b678 bootutil: Use boot_state_init instead of
    boot_state_clear
  - dd4b01f4 boot_serial: Initialize state with boot_state_init
  - a312656b bootutil: Add boot_loader_state_init
  - e9255183 bootutil: Allow using psa_key_id_t in AES crypto
    context
  - 3af40a31 boot: zephyr: sysflash: Increase number of supported
    images
  - c27bb0f4 boot_serial: Use struct enc_data in
    decrypt_region_inplace
  - 0287fd4d boot_serial: Fix double boot_loader_state init
  - f846e9e9 boot: zephyr: Allow disabling default multiple RAM
    region file
  - 0ccce2f7 bootutil: In boot_swap_image add return check from
    boot_read_enc_key
  - 32b3c18b bootutil: Refactor boot_read_enc_key

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>
(cherry picked from commit 881584112f75da8876f62d6e908b12bbd189b1bb)

valeriosetti committed:
[nrf fromtree] manifest: tf-m: align Mbed TLS header files to v3.6.5

Update manifest to align the copy of Mbed TLS header files which lives
in the TF-M repository to what's in Mbed TLS release v3.6.5.

Signed-off-by: Valerio Setti <vsetti@baylibre.com>
(cherry picked from commit e0fdb8792992c3b12f7bf13e629713fe80f7cf8b)

bjarki-andreasen commented on drivers/comparator/comparator_nrf_lpcomp.c:
```diff
@@ -134,99 +126,6 @@ static int shim_nrf_lpcomp_pm_callback(const struct device *dev, enum pm_device_
-
-#if NRF_GPIO_HAS_RETENTION_SETCLEAR
-	nrf_gpio_pin_retain_disable(shim_nrf_comp_ain_map[shim]);
-#endif
```
Has this made it into NRFX? without it AIN pins on H20 will not work.

jaz1-nordic committed:
[nrf noup] dts: nordic: remove leftover file

File no longer needed, can be removed.

Signed-off-by: Jakub Zymelka <jakub.zymelka@nordicsemi.no>

masz-nordic committed:
[nrf noup] boards: native: nrf_bsim: apply coding style

This is a subset of upstream's commit:
ccbc21bb32a873dd0480fd07c307acc2d51b1791
to prevent dependencies on other boards.

Signed-off-by: Marcin Szymczyk <marcin.szymczyk@nordicsemi.no>

masz-nordic committed:
[nrf noup] scripts: ci: check_compliance: fix whitespace

This is a subset of upstream's commit:
f2af86471220b4d269d006ad44c9a2bfe19f62aa
to prevent further dependencies.

Signed-off-by: Marcin Szymczyk <marcin.szymczyk@nordicsemi.no>

masz-nordic commented on modules/hal_nordic/nrfx/Kconfig:
```diff
@@ -1308,6 +1308,6 @@ endmenu
 
```
Done
